contract/OffchainAggregatorBilling.sol

```diff
@@ -167,7 +167,63 @@ contract OffchainAggregatorBilling is Owned {
     }
     s_oracleObservationsCounts = counts;
     s_gasReimbursementsLinkWei = gas;
+  }
+
+  /*
+   * @notice emitted when the LINK token contract is set
+   * @param _oldLinkToken the address of the old LINK token contract
+   * @param _newLinkToken the address of the new LINK token contract
+   */
+  event LinkTokenSet(
+    LinkTokenInterface indexed _oldLinkToken,
+    LinkTokenInterface indexed _newLinkToken
+  );
 
+  /*
+   * @notice sets the LINK token contract used for paying oracles
+   * @param _linkToken the address of the LINK token contract
+   * @param _recipient remaining funds from the previous token contract are transfered
+   * here
+   * @dev this function will return early (without an error) without changing any state
+   * if _linkToken equals getLinkToken().
+   * @dev this will trigger a payout so that a malicious owner cannot take from oracles
+   * what is already owed to them.
+   * @dev we assume that the token contract is correct. This contract is not written
+   * to handle misbehaving ERC20 tokens!
+   */
+  function setLinkToken(
+    LinkTokenInterface _linkToken,
+    address _recipient
+  ) external
+    onlyOwner()
+  {
+    LinkTokenInterface oldLinkToken = s_linkToken;
+    if (_linkToken == oldLinkToken) {
+      // No change, nothing to be done
+      return;
+    }
+    // call balanceOf as a sanity check on whether we're talking to a token
+    // contract
+    _linkToken.balanceOf(address(this));
+    // we break CEI here, but that's okay because we're dealing with a correct
+    // token contract (by assumption).
+    payOracles();
+    uint256 remainingBalance = oldLinkToken.balanceOf(address(this));
+    require(oldLinkToken.transfer(_recipient, remainingBalance), "transfer remaining funds failed");
```

Comment:
`"remaining-funds transfer failed"` stays within the 32-byte limit, and is a bit easier to parse grammatically.